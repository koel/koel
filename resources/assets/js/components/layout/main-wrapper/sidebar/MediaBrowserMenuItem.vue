<template>
  <SidebarItem :href screen="MediaBrowser">
    <template #icon>
      <Icon :icon="faFolderTree" fixed-width />
    </template>
    Media Browser
  </SidebarItem>
</template>

<script setup lang="ts">
import { faFolderTree } from '@fortawesome/free-solid-svg-icons'

import { ref } from 'vue'
import { useRouter } from '@/composables/useRouter'

import SidebarItem from '@/components/layout/main-wrapper/sidebar/SidebarItem.vue'

const { onRouteChanged, url } = useRouter()

const href = ref(url('media-browser'))

// Update the href when the route changes, so that clicking on the item will go to the previously visited folder.
onRouteChanged(handler => {
  if (handler.screen !== 'MediaBrowser') {
    return
  }

  href.value = url('media-browser', handler.params)
})
</script>

<style scoped lang="postcss">

</style>
