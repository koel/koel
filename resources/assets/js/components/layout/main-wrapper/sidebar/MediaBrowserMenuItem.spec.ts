import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import MediaBrowserMenuItem from './MediaBrowserMenuItem.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => {
      expect(this.render(MediaBrowserMenuItem).html()).toMatchSnapshot()
    })

    it('keeps track of the active path', async () => {
      const { html } = this.render(MediaBrowserMenuItem)

      await this.router.activateRoute({
        path: '_',
        screen: 'MediaBrowser',
      }, {
        path: 'foo/bar',
      })

      await this.tick(2)

      expect(html()).toMatchSnapshot()
    })
  }
}
