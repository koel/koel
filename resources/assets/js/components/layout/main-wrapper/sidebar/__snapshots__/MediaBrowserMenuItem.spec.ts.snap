// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mediaBrowserMenuItem.vue > keeps track of the active path 1`] = `<li data-v-7589e0e3="" class="relative before:right-0 px-6 before:top-1/4 before:w-[4px] before:h-1/2 before:absolute before:rounded-full before:transition-[box-shadow,_background-color] before:ease-in-out before:duration-500" data-testid="sidebar-item" screen="MediaBrowser"><a data-v-7589e0e3="" href="/#/browse" class="flex items-center overflow-x-hidden gap-3 h-11 relative active:pt-0.5 active:pr-0 active:pb-0 active:pl-0.5 text-k-fg-70 hover:text-k-fg"><span data-v-7589e0e3=""><br data-testid="Icon" icon="[object Object]" fixed-width=""></span><span data-v-7589e0e3="" class="overflow-hidden text-ellipsis whitespace-nowrap"> Media Browser </span></a></li>`;

exports[`mediaBrowserMenuItem.vue > renders 1`] = `<li data-v-7589e0e3="" class="relative before:right-0 px-6 before:top-1/4 before:w-[4px] before:h-1/2 before:absolute before:rounded-full before:transition-[box-shadow,_background-color] before:ease-in-out before:duration-500" data-testid="sidebar-item" screen="MediaBrowser"><a data-v-7589e0e3="" href="/#/browse" class="flex items-center overflow-x-hidden gap-3 h-11 relative active:pt-0.5 active:pr-0 active:pb-0 active:pl-0.5 text-k-fg-70 hover:text-k-fg"><span data-v-7589e0e3=""><br data-testid="Icon" icon="[object Object]" fixed-width=""></span><span data-v-7589e0e3="" class="overflow-hidden text-ellipsis whitespace-nowrap"> Media Browser </span></a></li>`;
